dotnet/src/SemanticKernel.Abstractions/Services/OrderedAIServiceSelector.cs

```diff
@@ -52,6 +53,13 @@ internal sealed class OrderedAIServiceSelector : IAIServiceSelector
                         return true;
                     }
                 }
+            }
+
+            // Search by model id next
+            foreach (var keyValue in executionSettings)
```

Comment:
Two loops in the same thing, see if we can use the same loop. 

Check this in the first
``` 
if (string.IsNullOrWhiteSpace(settings.ModelId))
```

Reply:
It was previously one loop but Matthew wants it split to be two so we search by service id first and only if that fails do we try by model id.